lstchain/visualization/plot_calib.py

```diff
@@ -49,297 +49,303 @@ def read_file(file_name, tel_id=1):
-        #charge_median = ff_data.charge_median[chan]
-        charge_mean = ff_data.charge_mean[chan]
-        charge_std = ff_data.charge_std[chan]
-        #median_ped = ped_data.charge_median[chan]
-        mean_ped = ped_data.charge_mean[chan]
-        ped_std = ped_data.charge_std[chan]
-
-        # select good pixels
-        select = np.logical_not(mask[chan])
-        fig = plt.figure(chan+10, figsize=(12, 18))
-        fig.tight_layout(rect=[0, 0.0, 1, 0.95])
-
-        fig.suptitle(f"Run {run} channel: {channel[chan]}", fontsize=25)
-
-        # charge
-        plt.subplot(321)
-        plt.tight_layout()
-        median = int(np.median(charge_mean[select]))
-        rms = np.std(charge_mean[select])
-        label=f"Median {median:3.2f}, std {rms:5.0f}"
-        plt.xlabel('charge (ADC)', fontsize=20)
-        plt.ylabel('pixels', fontsize=20)
-        plt.hist(charge_mean[select], bins=50,label=label)
-        plt.legend()
-
-        plt.subplot(322)
-        plt.tight_layout()
-        plt.ylabel('pixels', fontsize=20)
-        plt.xlabel('charge std', fontsize=20)
-        median = np.median(charge_std[select])
-        rms = np.std(charge_std[select])
-        label=f"Median {median:3.2f}, std {rms:3.2f}"
-        plt.hist(charge_std[select], bins=50,label=label)
-        plt.legend()
-
-        # pedestal charge
-        plt.subplot(323)
-        plt.tight_layout()
-        plt.ylabel('pixels', fontsize=20)
-        plt.xlabel('pedestal', fontsize=20)
-        median = np.median(mean_ped[select])
-        rms = np.std(mean_ped[select])
-        label=f"Median {median:3.2f}, std {rms:3.2f}"
-        plt.hist(mean_ped[select], bins=50,label=label)
-        plt.legend()
-
-        # pedestal std
-        plt.subplot(324)
-        plt.ylabel('pixels', fontsize=20)
-        plt.xlabel('pedestal std', fontsize=20)
-        median = np.median(ped_std[select])
-        rms = np.std(ped_std[select])
-        label=f"Median {median:3.2f}, std {rms:3.2f}"
-        plt.hist(ped_std[select], bins=50,label=label)
-        plt.legend()
-
-        # relative gain
-        plt.subplot(325)
-        plt.tight_layout()
-        plt.ylabel('pixels', fontsize=20)
-        plt.xlabel('relative signal', fontsize=20)
-        median = np.median(gain_median[select])
-        rms = np.std(gain_median[select])
-        label=f"Relative gain {median:3.2f}, std {rms:5.2f}"
-        plt.hist(gain_median[select], bins=50,label=label)
-        plt.legend()
-
-        # photon electrons
-        plt.subplot(326)
-        plt.tight_layout()
-        plt.ylabel('pixels', fontsize=20)
-        plt.xlabel('pe', fontsize=20)
-        median = np.median(n_pe[select])
-        rms = np.std(n_pe[select])
-        label=f"Median {median:3.2f}, std {rms:3.2f}"
-        plt.hist(n_pe[select], bins=50,label=label)
-        plt.legend()
-        plt.subplots_adjust(top=0.92)
-        if plot_file != "none":
-            pp.savefig(plt.gcf())
-
-
-    if plot_file != "none":
```

Comment:
is this condition removed?

Reply:
You only open the `PDFPages` `if plot_file is not None` and the `with` context manager closes it afterward. I just close the plt object right after saving the figure.